lib/src/lints/reverse_if_to_reduce_nesting/reverse_if_to_reduce_nesting_rule.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:analyzer/error/listener.dart';
+import 'package:custom_lint_builder/custom_lint_builder.dart';
+import 'package:solid_lints/src/lints/reverse_if_to_reduce_nesting/visitors/reverse_if_to_reduce_nesting_visitor.dart';
+import 'package:solid_lints/src/models/rule_config.dart';
+import 'package:solid_lints/src/models/solid_lint_rule.dart';
+
+/// A `reverse_if_to_reduce_nesting` rule which promotes using guard clauses
+/// over nesting `if` statements
+///
+/// ### Example
+///
+/// #### BAD:
+///
+/// ```dart
+/// void func() {
+///   if(a){ //LINT
+///     if(b){ //LINT
+///       c;
+///     }
+///   }
+/// }
+/// ```
+///
+/// #### GOOD:
+///
+/// ```dart
+/// void func() {
+///   if(!a) return;
+///   if(!b) return;
+///   c;
+/// }
+/// ```
+class ReverseIfToReduceNestingRule extends SolidLintRule {
+  /// The [LintCode] of this lint rule that represents the error if
+  /// 'if' statements should be reversed
+  static const String lintName = 'reverse_if_to_reduce_nesting';
```

Comment:
A `guard` clause unfortunately has an [unconventional meaning in Dart,](https://dart.dev/language/branches#guard-clause) so we can't use that.
Also, not sure we need to put an accent on nesting here. Maybe something like this?
```suggestion
  static const String lintName = 'prefer_early_return';
```